builder/docker.go

```diff
@@ -163,21 +164,35 @@ func (b *Builder) getDockerMounts() ([]string, error) {
 // RunCommandInContainer will pull the content necessary to build a docker
 // image capable of running the desired command, build that image, and then
 // run the command in that image.
-func (b *Builder) RunCommandInContainer(cmd []string) error {
-	format, err := b.getUpstreamFormat(b.UpstreamVer)
-	if err != nil {
-		return err
+func (b *Builder) RunCommandInContainer(cmd []string, offline bool) error {
+	var imageName string
+	var format string
+	var err error
+	if !offline {
+		format, err = b.getUpstreamFormat(b.UpstreamVer)
+		if err != nil {
+			return err
+		}
+	} else {
+		var hostFormat []byte
+		hostFormat, err = ioutil.ReadFile("/usr/share/defaults/swupd/format")
+		if err != nil && !os.IsNotExist(err) {
```

Comment:
If I'm reading this correctly, this block is entered when any error is raised other than the file not existing. If the file doesn't exist, then no content is read, so using the `hostFormat` value makes little sense. So, why specify `!os.IsNotExist(err)` here?

Reply:
Elsewhere in `docker.go` we have [`GetHostAndUpstreamFormats`](https://github.com/clearlinux/mixer-tools/blob/master/builder/docker.go#L34).

a) It'd be nice if these two methods used the same code snippet to parse the format file.
b) In that method, `!os.IsNotExist(err)` makes sense, because the file not existing isn't an error condition, but a warning -- the code that calls it checks for the returned empty string and warns that the format could not be determined. Here, yes, it seems like you want to error out if it cannot be determined. In that case, though, drop the `&& os.IsNotExist(err)` altogether. Otherwise you're _only_ catching File Not Exist.